pages/auth/signin/signin.js

```diff
@@ -1,21 +1,61 @@
 "use strict";
 
-const email = document.getElementById('email');
-const password = document.getElementById('password');
-const signinButton = document.querySelector('.signin_button');
+import {
+  emailInput,
+  emailErrorMsg,
+  passwordInput,
+  passwordErrorMsg,
+  signinButton,
+  togglePassword,
+} from "../../../modules/form-validation.js";
+import { ERROR_MSG, checkRegex, togglePwd } from "../../../modules/form-validation.js";
 
-function checkForm () {
-  const emailRegex = /^[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*\.[a-zA-Z]{2,3}/;
-  const passwordRegEx = /^[A-Za-z0-9]{8,20}$/;
-  
-  const isEmailValid = emailRegex.test(email.value);
-  const isPasswordValid = passwordRegEx.test(password.value);
+// 에러 메세지 초기화
+emailErrorMsg.innerText = '';
+passwordErrorMsg.innerText = '';
 
-  const isFormValid = isEmailValid && isPasswordValid;
+// 에러 메세지 반환 검사
+const checkValidation = (target, msgTarget) => {
+  const isValid = checkRegex(target);
+  const targetId = target.id;
+  const capitalizedTargetId = targetId.charAt(0).toUpperCase() + targetId.slice(1);
 
-  signinButton.disabled = !isFormValid;
-  signinButton.classList.toggle('btn_abled', isFormValid);
+  if (isValid ==='required') {
+    // 아무것도 입력되지 않은 경우
+    target.classList.add('border_red');
+    msgTarget.innerText = ERROR_MSG['required' + capitalizedTargetId];
+  } else if (isValid === 'invalid' + capitalizedTargetId) {
+    // 유효성 검사 불통과
+    target.classList.add('border_red');
+    msgTarget.innerText = ERROR_MSG['invalid' + capitalizedTargetId];
+  } else {
+    // 모든 유효성 검사 통과
+    target.classList.remove('border_red');
+    msgTarget.innerText = '';
+  }
+  return isValid;
 };
 
-email.addEventListener('input', checkForm);
-password.addEventListener('input', checkForm);
\ No newline at end of file
+// 폼 유효성 검사
+const checkFormValidation = () => {
+  const isEmailValid = emailInput.value !== '' && checkValidation(emailInput, emailErrorMsg);
+  const isPasswordValid = passwordInput.value !== '' && checkValidation(passwordInput, passwordErrorMsg);
+
+  if (
+    isEmailValid === true && 
+    isPasswordValid === true
+  ) {
+    signinButton.disabled = false;
+  } else {
+    signinButton.disabled = true;
+  }
+};
+
+const itemsPage = () => {
+  window.location.href = 'items.html';
+};
```

Comment:
이렇게 만들어두는 편이 좋을거 같아요.
goto 또는 push로 이름을 붙일 수 있는데, 각종 프레임워크/라이브러리에서도 익숙한 함수입니다. 

```suggestion
const goto = (url) => {
  window.location.href = url
};

// 사용하는 곳에서
goto('items.html')
```